rfd/xxxx-machine-id-anonymous-telemetry.md

```diff
@@ -0,0 +1,89 @@
+---
+authors: Noah Stride (noah.stride@goteleport.com)
+state: draft
+---
+
+# RFD XXXX - Machine ID Anonymous Agent Telemetry
+
+## Required Approvers
+
+* Engineering: @zmb3
+* Product: @klizhentas && @xinding33
+
+## What
+
+Collect anonymous usage information about invocations of the Machine ID agent
+from the agent itself.
+
+Out of scope is the collection of usage information from the Auth server.
+
+## Why
+
+It is currently difficult to determine the uptake of Machine ID by customers,
+and for what use-cases they use Machine ID for. This makes product decisions
+about determining areas of focus more difficult.
+
+Basic anonymous telemetry from the `tbot` agent will provide helpful
+information on adoption without compromising privacy.
+
+## Details
+
+### Collecting events
+
+
+### Submitting events
+
+### Consent
+
+Users will explicitly opt-in to anonymous usage telemetry with a command-line
+parameter or configuration file field e.g:
```

Comment:
What about environment variable? Seems like all CI/CD providers will support this, and it means you just have to delete an entire line to disable telemetry, rather than edit the line with the `tbot` invocation.

Reply:
I'd agree that environment variables are probably universal enough - as far as I can tell we don't have any other environment variable configuration for `tbot`, but it wouldn't be difficult to just implement this as a one-off `Getenv` especially as this is separate from the core of `tbot` itself.

Reply:
5bf8689df5

I've gone for `TBOT_ANONYMOUS_TELEMETRY=1` - but maybe we should consider making this a universal environment variable across all clients and naming it `TELEPORT_ANONYMOUS_TELEMETRY=1` !